### ISSUE TITLE
V2 internal OIDC redirect loop

### ISSUE THREAD

**Issue by ddsmith2-eprod** [coder/coder#6300]
Hello,

I have a working v1 cluster that uses an internal OIDC server to authenticate.  We are setting up v2 and running into an infinite redirect loop. The settings are fairly basic.  At first, we were getting an authorization denied, so I had our administrator whitelist the callback URL.  

Now when we login we get a message about revoking permissions.  After clicking ok we get a flashing webpage that appears to be stuck in a redirect loop.  After a minute or so we get throttled from the OIDC API for too many requests in 60 seconds.

I'm not seeing any errors from the Coder pod and I'm not sure where else to check for the cause of the issue.

```
    - name: CODER_ACCESS_URL
      value: "https://coder.pdds2.mydomain.com"
    - name: CODER_OIDC_ISSUER_URL
      value: "https://myportal.mydomain.com/SecureAuth92"
#   - name: CODER_OIDC_EMAIL_DOMAIN
#     value: "mydomain.com"
    - name: CODER_OIDC_CLIENT_ID
      value: "abcdefghijkl12456"
    - name: CODER_OIDC_CLIENT_SECRET
      value: "<REDACTED>"
    - name: CODER_OIDC_SCOPES
      value: "openid,profile,email,groups"
```

I checked our v1 cluster and the setup looks pretty much identical on the settings.  mydomain has replaced with our actual domain name.

**Comment by MrPeacockNLB** [comment]
just yesterday configured OIDC with Azure AD as oauth2 provider. Everything works fine. If I activate group syncing then I get the described redirect loop. As a first mitigation try to disable group syncing and remove groups scope or remove completely CODER_OIDC_SCOPES

**Comment by ddsmith2-eprod** [comment]
Thanks for the reply!  I tried removing the scopes and now I get an authorization error.  I'm reaching out to our OIDC admin to see about adding authorization.

```
{"message":"Failed to obtain user information claims.","detail":"The OIDC provider returned no claims as part of the `id_token`. The attempt to fetch claims via the UserInfo endpoint failed: 401 Unauthorized: "}
```

**Comment by ddsmith2-eprod** [comment]
We tried creating a brand new OIDC client ID.  We get the same behavior; however, we did find that if we comment the CODER_OIDC_SCOPES from our values.yaml we can type a username into Secure Auth and get logged in (account is created in Coder v2).  If I re-add scopes I get a similar error to above with 401 errors.  We kind of have it working now except our end users will have to type in their username instead of the normal automatic login.

If anyone knows what changes, we should make, to get scopes fully working it would be greatly appreciated.
